plugins/dataset_feature_space.cc

```diff
@@ -114,6 +114,9 @@ getColumnInfo(std::shared_ptr<Dataset> dataset,
                 auto categorical = std::make_shared<ML::Fixed_Categorical_Info>(allValues);
                 result.info = ML::Feature_Info(categorical);
             }
+            else if (types.size() == 1 && types.begin()->first == CellValue::EMPTY) {
```

Comment:
As for the case of a constant int, you are right, we would then keep it. If I keep a very narrow view, the current issue is about dealing with NULLs.

We could open a separate ticket to deal with non null features of cardinality one, but they *should* be less problematic as they *should* be of the right type. (Vs null that defaults to numerical which is impossible to determine as correct or wrong.)